src/commands.ts

```diff
@@ -547,3 +550,155 @@ export function registerCommands(context: vscode.ExtensionContext, prManager: Pu
 		}
 	}));
 }
+
+function sanitizeRepositoryName(value: string): string {
+	return value.trim().replace(/[^a-z0-9_.]/ig, '-');
+}
+
+export function registerGlobalCommands(context: vscode.ExtensionContext, gitAPI: GitAPI, credentialStore: CredentialStore) {
+	// TODO: join with other code from https://github.com/microsoft/vscode-pull-request-github/pull/1661
+	async function getHub(): Promise<GitHub | undefined> {
+		if (await credentialStore.hasOctokit()) {
+			return await credentialStore.getHub()!;
+		}
+
+		const hub = await credentialStore.loginWithConfirmation();
+
+		if (!hub) {
+			return credentialStore.login();
+		}
```

Comment:
@RMacfarlane You mean something like this?

```suggestion
		return credentialStore.login();
```

Reply:
Something like this:

```ts
		if (await credentialStore.hasOctokit()) {
			return await credentialStore.getHub()!;
		} else {
                        return await credentialStore.login();
                }
```